File: ElisBackend/Core/Domain/Entities/Stock.cs
using ElisBackend.Core.Domain.Abstractions;

namespace ElisBackend.Core.Domain.Entities;

// Stock description
public class Stock(string name, string isin, string instrumentCode, Exchange exchange, Currency currency) : IStock
{
    public string Name { get; private set; } = name;
    public string Isin { get; private set; } = isin;
    public string InstrumentCode { get; private set; } = instrumentCode;
    public IExchange Exchange { get; private set; } = exchange;
    public ICurrency Currency { get; private set; } = currency;
}

File: GetStockPrizes/Core/Entities/Stock.py
class Stock:
    def __init__(self):
        self.isin = ''
        self.name = ''
        self.exchange = ''
        self.instrumentcode = ''

    def SetStock(self, isin, name, exchange, instrumentcode):
        self.isin = isin
        self.name = name
        self.exchange = exchange
        self.instrumentcode = instrumentcode
